### ISSUE TITLE
Several Questions about the Paper Content

### ISSUE THREAD

**Issue by 2502128021** [Fantasy-AMAP/fantasy-talking#3]
Hi，
 I have several questions about the paper content, and hope your answer.
1.In your paper, under the section "Dual-Stage Audio-Visual Alignment," it mentions two stages of training. Do these two stages correspond to the same audio cross-attention layer, or are there two separate layers? It seems from the description in the paper that there is only one layer. Why not directly use Frame-Level Training?
2.During the forward pass, you mentioned that the CFG (Classifier-Free Guidance) for audio is set to 4.5. When performing the forward inference, is the unconditional generation part also conditioned on text, or is the text condition left empty?

**Comment by wangqiang9** [comment]
Hi, @2502128021 .
Regarding the first question, the two stages correspond to the same cross-attention layer. Additionally, the ablation study at the frame-level is presented in Chapter 5 of the paper. 
As for the second question, our CFG strategy is as follows:
```
noise_pred = noise_pred_uncond + text_cfg_scale * (noise_pred_drop_audio - noise_pred_uncond) + audio_cfg_scale * (noise_pred_cond - noise_pred_drop_audio)
```

**Comment by 2502128021** [comment]
thx for your reply! follow your cfg strategy，we need to set batch size=3, is that right? One for unconditional, another for text only, and the last one for text and audio together? So what value should we set for text_cfg_scale?

发自网易邮箱 iPhone 版
---- Replied Message ----
| From | Wang ***@***.***> |
| Date | 04/14/2025 14:34 |
| To | Fantasy-AMAP/fantasy-talking ***@***.***> |
| Cc | 2502128021 ***@***.***>,
Mention ***@***.***> |
| Subject | Re: [Fantasy-AMAP/fantasy-talking] Several Questions about the Paper Content (Issue #3) |

Hi, @2502128021 .
Regarding the first question, the two stages correspond to the same cross-attention layer. Additionally, the ablation study at the frame-level is presented in Chapter 5 of the paper.
As for the second question, our CFG strategy is as follows:

noise_pred = noise_pred_uncond + text_cfg_scale * (noise_pred_drop_audio - noise_pred_uncond) + audio_cfg_scale * (noise_pred_cond - noise_pred_drop_audio)


—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***>

wangqiang9 left a comment (Fantasy-AMAP/fantasy-talking#3)

Hi, @2502128021 .
Regarding the first question, the two stages correspond to the same cross-attention layer. Additionally, the ablation study at the frame-level is presented in Chapter 5 of the paper.
As for the second question, our CFG strategy is as follows:

noise_pred = noise_pred_uncond + text_cfg_scale * (noise_pred_drop_audio - noise_pred_uncond) + audio_cfg_scale * (noise_pred_cond - noise_pred_drop_audio)

**Comment by noorkaximi** [comment]
when we can expect the source code?

**Comment by 2502128021** [comment]
hello，I have two more questions, and hope for your reply.
q1: Is it feasible to directly conduct training at the frame-level stage?
q2: Is positional encoding needed for the audio part? If so, what kind of positional encoding should be added at the clip-level and frame-level stages?

**Comment by xilanhua12138** [comment]
Hi！ When training what kind of noise scheduler do you use? uniform sampling or sample logit-normal with a shift?
